src/main/java/bio/terra/workspace/service/create/flight/CreateWorkspaceAuthzStep.java

```diff
@@ -0,0 +1,37 @@
+package bio.terra.workspace.service.create.flight;
+
+import bio.terra.stairway.FlightContext;
+import bio.terra.stairway.FlightMap;
+import bio.terra.stairway.Step;
+import bio.terra.stairway.StepResult;
+import bio.terra.stairway.exception.RetryException;
+import bio.terra.workspace.service.iam.AuthenticatedUserRequest;
+import bio.terra.workspace.service.iam.SamService;
+import java.util.UUID;
+
+public class CreateWorkspaceAuthzStep implements Step {
+
+  private SamService samService;
+  private AuthenticatedUserRequest userReq;
+
+  public CreateWorkspaceAuthzStep(SamService samService, AuthenticatedUserRequest userReq) {
+    this.samService = samService;
+    this.userReq = userReq;
+  }
+
+  @Override
+  public StepResult doStep(FlightContext flightContext) throws RetryException {
+    FlightMap inputMap = flightContext.getInputParameters();
+    UUID workspaceID = inputMap.get(WorkspaceFlightMapKeys.WORKSPACE_ID, UUID.class);
+    samService.createWorkspaceWithDefaults(userReq.getRequiredToken(), workspaceID);
+    return StepResult.getStepResultSuccess();
+  }
+
+  @Override
+  public StepResult undoStep(FlightContext flightContext) {
+    FlightMap inputMap = flightContext.getInputParameters();
+    UUID workspaceID = inputMap.get(WorkspaceFlightMapKeys.WORKSPACE_ID, UUID.class);
+    samService.deleteWorkspace(userReq.getRequiredToken(), workspaceID);
+    return StepResult.getStepResultSuccess();
+  }
+}
```

Comment:
Probably more of a question about stairway: does stairway provide all the logging we want/need to debug problems? IF not, add some logging - and potentially error-trapping in order to generate log messages - to these steps.

Reply:
Each stairway step (both forward and backwards) logs information when it starts or hits any exceptions - I think that's a reasonable level of logging for now, but I'm fine adding more logging if you think it'd be helpful